### PR TITLE
utils.py output generator and skycoord.rst display modifications

### LINKED ISSUE #9321: Automate generation of the `SkyCoord` tab-completion documentation
The documentation for `SkyCoord` has a section listing the attributes on a `SkyCoord` (http://docs.astropy.org/en/stable/coordinates/skycoord.html#attributes, which is in `docs/coordinates/skycoord.rst`).  This could/should be auto-generated instead of the current state of it having to be manually updated.

It looks to me like the tab-completion list is the same as you get from the following:
```
>>> from astropy.coordinates import SkyCoord
>>> sc = SkyCoord(1*u.deg, 2*u.deg)
>>> ["sc."+attrname for attrname in dir(sc) if not attrname.startswith('_')]
```
so in principle, it's just a matter of injecting that list into the documentation.  Would be nice to do this so that it's not necessary to manually keep that list up to date (or more likely: forget to do so...).

cc @adrn (who recently pointed out this table isn't up-to-date)

### PR DESCRIPTION
**Pull request by AiIuru5**
### Description

This pull request is to address ...

Fixes #9321

This uses a utils files to generate the python output, which is added to skycoord.rst through .. include: txt. I'm not sure if this is the best idea, but it is working and i wanted to see if it would be any help. Method 2 in the discussion under the pull request. I didn't manage to duplicate the formatting in the section. 
image

### CODE REVIEW

File: astropy/coordinates/utils.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    try:
+        from astropy.coordinates import SkyCoord
+        import astropy.units as u
+        sc = SkyCoord(1*u.deg, 2*u.deg)
+        output =' \n'.join(["sc."+attrname for attrname in dir(sc) if not attrname.startswith('_')])
+
+    except Exception as e:
+        output = str(e)
+
+    with open('../../docs/coordinates/TabCompleteList.txt', 'w') as rst_file:
```
Comment (pllim) [inline astropy/coordinates/utils.py:12]: Assuming we want to do it this way, we'll have to also ask git to ignore this file.

**Comment by AiIuru5** [comment]
I assume you mean addding Ailuru5-main\docs\coordinates\TabCompleteList.txt to the .gitignore file?

**Comment by AiIuru5** [comment]
I found a sphinx extension that would allow us to keep the code in the rst file, which seems like the cleanest solution, but i wouldn't know how to include that in the project. I can look into it, i just don't know where to start.

**Comment by pllim** [comment]
Hmm somehow your changes are failing the test.

**Comment by AiIuru5** [comment]
the doc tests seem to be failing on a  portion of the code that i didn't modify  also in skycoords rst. I'm confused. Would it help to import the changes i made to a more updated branch on my repository and then submit another pull request? I assume this works in the astropy/astropy and i'm just behind in versions or something. 

Expected:
    <Longitude 1. deg>
Got:
    <Longitude [  0.,  36.,  72., ..., 288., 324.,   0.] deg>

**Comment by pllim** [comment]
Not sure how stale your branch is, but you can try to rebase:

```
git fetch upstream main
git rebase upstream/main
```

**Comment by AiIuru5** [comment]
it says it's up to date? it didn't refresh the checking software.

**Comment by pllim** [comment]
I do not see this failure on `main`, so it must have come from this PR. Maybe something in the new RST is messing with the state of variables that affect our doctest.

**Comment by pllim** [comment]
Thanks but we have decided to not pursue this after all. Feel free to find another issue to work on. 🙇‍♀️

**Pull request closed without merging**